int32 target_number
float32 period
---
int32 reached_number
---
int32 current_number